variables.tf

```
@@ -18,6 +18,18 @@ variable "sku" {
   default     = "Basic"
 }
 
+variable "webhook" {
+  description = "A list of objects required for this Webhook."
+  type = list(object({
+    name        = string
+    service_uri = string
+    status      = string
+    scope       = string
+    actions     = list(string)
+  }))
+  default = []
+}
```

Update variable name and description to better reflect that you can create multiple webhooks, and set type to map of objects and make some properties optional with default values:

```suggestion
variable "webhooks" {
  description = "A map of webhooks to create for this Container Registry."
  type = map(object({
    name        = string
    service_uri = string
    actions     = optional(list(string), ["push"])
    status      = optional(string, "enabled")
    scope       = optional(string, "")
  }))
  default = {}
```

By using maps, you could create two webhhooks `web` and `sql` using the following syntax:

```terraform
module "acr" {
  # source and arguments

  webhooks = {
    "web" = {
      # webhook properties
    }
    "sql" = {
      # webhook properties
    }
  }
}
```

Terraform would then create two webhook resources:

1. `module.acr.azurerm_container_registry_webhook.web`
2. `module.acr.azurerm_container_registry_webhook.sql`

This way, you have full control over the names of resources in your Terraform state file.

This is the same way we've done it for the other modules 🙂